### PR TITLE
dd: move tests of slow reader to test_dd.rs

### PR DESCRIPTION
**Pull request by jfinkels**
Move some tests that simulate a slow reader from `dd.rs` to `tests/by-util/test_dd.rs`, and employ a FIFO and `sleep()` to simulate the slow reader instead of a custom struct that implements `Read`. This change restricts the type of `Input`s the `Output::dd_out()` function can accept, facilitating a future change to make `Input` an enum.

Inspired by https://github.com/uutils/coreutils/pull/4134#issuecomment-1315967939

### CODE REVIEW

File: tests/by-util/test_dd.rs [hunk 1]
```diff
@@ -5,6 +5,8 @@ use crate::common::util::*;
 use std::fs::{File, OpenOptions};
 use std::io::{BufReader, Read, Write};
 use std::path::PathBuf;
+use std::thread::sleep;
```
Comment (sylvestre) [inline tests/by-util/test_dd.rs:8]: a warning here :)

**Review by tertsdiepraam — COMMENTED** [review]
This is great! `dd` has always felt too complicated and I'm excited to see it cleaned up once we can convert `Input` to an enum :)

File: tests/by-util/test_dd.rs (hunk 1 above)
Comment (jfinkels) [reply tests/by-util/test_dd.rs:9]: should be fixed now, thanks

File: tests/by-util/test_dd.rs [hunk 2]
```diff
@@ -5,6 +5,9 @@ use crate::common::util::*;
 use std::fs::{File, OpenOptions};
 use std::io::{BufReader, Read, Write};
 use std::path::PathBuf;
+#[cfg(not(windows))]
+use std::thread::sleep;
+use std::time::Duration;
```
Comment (sylvestre) [inline tests/by-util/test_dd.rs:10]:
```suggestion
#[cfg(not(windows))]
use std::time::Duration;
```

**Comment by jfinkels** [comment]
Sorry, one more update: I updated this branch to remove some test fixtures that are no longer needed. They were stored in `src/uu/dd/test-resources/`.

**Review by tertsdiepraam — APPROVED** [review]
Cool! Can be merged in my opinion if the CI is green

**Pull request merged**